agents/ahv/fence_ahv.py

```diff
@@ -0,0 +1,608 @@
+            raise AHVFenceAgentException from err
+
+        if current_power_state.lower() == power_state.lower():
+            logging.info("VM already powered %s", power_state.lower())
+            return
+
+        if power_state.lower() == 'on':
+            resp = self._power_on_vm(vm_uuid)
+        elif power_state.lower() == 'off':
+            resp = self._power_off_vm(vm_uuid)
+
+        task_id = resp.json()['data']['extId']
+
+        try:
+            self._wait_for_task(task_id, timeout)
+        except AHVFenceAgentException as err:
+            logging.error("Failed to power %s VM", power_state.lower())
+            logging.error("VM power %s task failed with status, %s",
+                            power_state.lower(), status)
+            raise AHVFenceAgentException from err
+        except TaskTimedOutException as err:
+            logging.error("Timed out powering %s VM %s",
+                          power_state.lower(), vm_uuid)
+            raise TaskTimedOutException from err
+
+        logging.info("Powered %s VM, %s successfully",
+                     power_state.lower(), vm_uuid)
+
+    def power_cycle_vm(self, vm_name=None, vm_uuid=None, timeout=None):
+        resp = None
+        status = None
+
+        if not timeout:
+            timeout = MIN_TIMEOUT
+
+        if not vm_name and not vm_uuid:
+            logging.error("Require at least one of VM name or VM UUID")
+            raise AHVFenceAgentException("No arguments provided")
+
+        if not vm_uuid:
+            vm_uuid = self._get_vm_uuid(vm_name)
+
+        resp = self._power_cycle_vm(vm_uuid)
+        task_id = resp.json()['data']['extId']
+
+        try:
+            self._wait_for_task(task_id, timeout)
+        except AHVFenceAgentException as err:
+            logging.error("Failed to power-cycle VM %s", vm_uuid)
+            logging.error("VM power-cycle task failed with status, %s", status)
+            raise AHVFenceAgentException from err
+        except TaskTimedOutException as err:
+            logging.error("Timed out power-cycling VM %s", vm_uuid)
+            raise TaskTimedOutException from err
+
+
+        logging.info("Power-cycled VM, %s", vm_uuid)
+
+
+def connect(options):
+    host = options["--ip"]
+    username = options["--username"]
+    password = options["--password"]
+    verify_ssl = True
+    disable_warnings = False
+
+    if "--ssl-insecure" in options:
+        verify_ssl = False
+        disable_warnings = True
+
+    client = NutanixV4Client(host, username, password,
+                             verify_ssl, disable_warnings)
+
+    try:
+        client.list_vms(limit=1)
+    except AHVFenceAgentException as err:
+        logging.error("Connection to Prism Central Failed")
+        logging.error(err)
+        fail(EC_LOGIN_DENIED)
+
+    return client
+
+def get_list(client, options):
+    vm_list = None
+    limit = None
+    filter_str =  None
+
+    if "--filter" in options:
```

Comment:
You can simplify these to `filter_str = options.get("--filter", None)`.

Reply:
Done.